### PR TITLE
Perl formatter improvement

### PR DESCRIPTION
**Pull request by jamadam**
The patch improves Perl code on non-ascii output. It reduces charactor corruption and well known warnings 'Wide character in print'.

It also change the literal delimiter from ""  to '' or q{} or q() to reduce variable escaping so that the test cases get more readable.

Since I haven't figure out how to build xpi so far, the patch is not properly tested.

### CODE REVIEW

**Comment by andreastt** [comment]
@samitbadle, did you have time to look at the bug you found?

**Comment by samitbadle** [comment]
I believe I fixed it, but need to merge the code. I will not have any time before this Friday.

**Comment by samitbadle** [comment]
Awaiting write access

**Comment by andreastt** [comment]
@samitbadle, consider yourself invited!

File: ide/plugins/perl-format/src/content/formats/perl-rc.js [hunk 1]
```diff
@@ -27,14 +27,12 @@ var formatter = this;
...
+		value = value.replace(/\'/g, '\\\'');
 		value = value.replace(/\r/g, '\\r');
 		value = value.replace(/\n/g, '\\n');
-		value = value.replace(/@/g, '\\@');
-		value = value.replace(/\$/g, '\\$');
-		return '"' + value + '"';
+		return "'" + value + "'";
 	} else {
-		return '""';
+		return "'";
```
Comment (samitbadle) [inline ide/plugins/perl-format/src/content/formats/perl-rc.js]: It should be return "''"; I will fix it

**Comment by samitbadle** [comment]
Thanks Andreas, now waiting for the clone, I need faster internet!

**Pull request merged**